.github/workflows/native-it-selected-graalvm.yml

```diff
@@ -0,0 +1,396 @@
+    steps:
+      - uses: actions/checkout@v4
+      - name: Calculate matrix from native-tests.json
+        id: calc-native-matrix
+        run: |
+          echo "GIB_IMPACTED_MODULES: ${GIB_IMPACTED_MODULES}"
+          json=$(.github/filter-native-tests-json.sh "${GIB_IMPACTED_MODULES}" | tr -d '\n')
+          # Remove Windows from the matrix
+          json=$(echo $json | jq 'del(.include[] | select(."os-name" == "windows-latest"))')
+          json=$(echo $json | tr -d '\n')
+          echo "${json}"
+          echo "matrix=${json}" >> $GITHUB_OUTPUT
+      - name: Calculate matrix from virtual-threads-tests.json
+        id: calc-virtual-threads-matrix
+        run: |
+          echo "GIB_IMPACTED_MODULES: ${GIB_IMPACTED_MODULES}"
+          json=$(.github/filter-virtual-threads-tests-json.sh "${GIB_IMPACTED_MODULES}" | tr -d '\n')
+          # Remove Windows from the matrix
+          json=$(echo $json | jq 'del(.include[] | select(."os-name" == "windows-latest"))')
+          json=$(echo $json | tr -d '\n')
+          echo "${json}"
+          echo "matrix=${json}" >> $GITHUB_OUTPUT      
+
+  virtual-thread-native-tests:
+    name: Native Tests - Virtual Thread - ${{matrix.category}} - ${{inputs.NATIVE_COMPILER}} ${{inputs.NATIVE_COMPILER_VERSION}} - ${{inputs.BRANCH}}
+    runs-on: ${{matrix.os-name}}
+    needs: [build-jdk17, calculate-test-jobs]
+    timeout-minutes: ${{matrix.timeout}}
+    strategy:
+      max-parallel: 12
+      fail-fast: false
+      matrix: ${{ fromJson(needs.calculate-test-jobs.outputs.virtual_threads_matrix) }}
+    steps:
+      - name: Gradle Enterprise environment
+        run: |
+          category=$(echo -n '${{matrix.category}}' | tr '[:upper:]' '[:lower:]' | tr -c '[:alnum:]-' '-' | sed -E 's/-+/-/g')
+          echo "GE_TAGS=virtual-thread-native-${category}" >> "$GITHUB_ENV"
+          echo "GE_CUSTOM_VALUES=gh-job-name=Native Tests - Virtual Thread - ${{matrix.category}}" >> "$GITHUB_ENV"
+      - uses: actions/checkout@v4
+      - name: Restore Maven Repository
+        uses: actions/cache/restore@v4
+        with:
+          path: ~/.m2/repository
+          # refresh cache every week to avoid unlimited growth
+          key: ${{ needs.build-jdk17.outputs.m2-cache-key }}
+      - name: Download .m2/repository/io/quarkus
+        uses: actions/download-artifact@v4
+        with:
+          name: m2-io-quarkus
+          path: .
+      - name: Extract .m2/repository/io/quarkus
+        run: tar -xzf m2-io-quarkus.tgz -C ~
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 21
+      - name: Setup GraalVM
+        id: setup-graalvm
+        uses: graalvm/setup-graalvm@v1
+        with:
+          java-version: ${{ inputs.NATIVE_COMPILER_VERSION }}
+          distribution: ${{ inputs.NATIVE_COMPILER }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
This one sets $JAVA_HOME to $GRAALVM_HOME by default making the setup-java step above redundant.

If you want to use JDK 21 as JAVA_HOME and only use GraalVM for the native build part add `set-java-home: 'false'` (see https://github.com/graalvm/setup-graalvm?tab=readme-ov-file#key-features)